src/app/store/effects/VehicleTestResultModel.effects.ts

```diff
@@ -33,9 +41,68 @@ export class VehicleTestResultModelEffects {
     )
   );
 
+  @Effect()
+  updateTestResult$ = this._actions$.pipe(
+    ofType<UpdateTestResult>(EVehicleTestResultModelActions.UpdateTestResult),
+    withLatestFrom(this._store.select(getVehicleTestResultModel)),
+    map(([{ testResultTestTypeNumber }, testResultsInState]): [
+      TestResultTestTypeNumber,
+      TestResultModel[]
+    ] => [testResultTestTypeNumber, testResultsInState]),
+    switchMap(([testResultTestTypeNumber, testResultsInState]) => {
+      const testResultTestTypeNumberObject = testResultTestTypeNumber;
+      let testResultObject: VehicleTestResultUpdate = {} as VehicleTestResultUpdate;
+      const { testResultUpdated } = testResultTestTypeNumber;
+
+      testResultObject = {
+        msUserDetails: { ...this.loggedUser.getUser() },
+        testResult: testResultUpdated
+      };
+
+      return this._testResultService
+        .updateTestResults(testResultObject.testResult.systemNumber, testResultObject)
+        .pipe(
+          switchMap((testRecordResult) => {
+            testResultTestTypeNumberObject.testResultUpdated = testRecordResult;
+            testResultTestTypeNumberObject.testResultsUpdated = this.updateTestResultsInState(
+              testResultsInState,
+              testResultTestTypeNumberObject
+            );
+            return [
+              new SetCurrentState(VIEW_STATE.VIEW_ONLY),
+              new UpdateTestResultSuccess(testResultTestTypeNumberObject),
+              new ClearErrorMessage()
+            ];
+          }),
+          catchError(({ error }) => {
+            const errorMessage = error.errors;
+            return [new SetErrorMessage(errorMessage)];
+          })
+        );
+    })
+  );
+
   constructor(
     private _testResultService: TestResultService,
-    private _store$: Store<IAppState>,
-    private _actions$: Actions
+    private _store: Store<IAppState>,
+    private _actions$: Actions,
+    private loggedUser: UserService
   ) {}
+
+  updateTestResultsInState(
+    testResultsInState: TestResultModel[],
+    testResultTestTypeNumberObject: TestResultTestTypeNumber
+  ): TestResultModel[] {
+    return testResultsInState.map((testResult) => {
```

Comment:
this can be extracted into a separate function for testability.
That will be function reused for test-result so maybe we can create a util file for it?
@alleypa @m-ujaffer thoughts on where to store the 'util' file?